msu/systems/persistent_data/persistent_data_system.nut

```diff
@@ -108,4 +110,46 @@
 		result += this.Separator
 		::logInfo(result);
 	}
+
+	function createFile( _fileName, _dataArray )
+	{
+		::MSU.requireInstanceOf(::MSU.Class.ArrayData, _dataArray);
+		this.validateFileName(_fileName);
+		local storage = ::PersistenceManager.createStorage(_fileName);
+		storage.beginWrite();
+		_dataArray.serialize(storage);
+		storage.endWrite();
+	}
+
+	function validateFileName( _fileName )
+	{
+		if (!this.FileNameRegexp.match(_fileName))
+		{
+			::logError("Battle Brothers file saves can only contain the characters 'a-zA-Z0-9_,-+# '");
+			throw ::MSU.Exception.InvalidValue(_fileName);
+		}
+	}
```

Comment:
Seems to be only used in that one spot and is not very long, so just move it into the function?

Reply:
@Enduriel

Reply:
@TaroEld do you mean the `FileNameRegexp`? I don't think we should be recreating it every function call if we can cache it

Reply:
I meant the whole validation function, but it doesn't really matter.